sdk/include/ndk/ketypes.h

```diff
@@ -1369,6 +1538,431 @@ typedef struct _KTHREAD
+#if (NTDDI_VERSION < NTDDI_WIN10)
+            ULONG KernelStackResident : 1;
+#else
+            ULONG AutoBoostActive : 1;
+#endif
+            ULONG ReadyTransition : 1;
+#if (NTDDI_VERSION < NTDDI_WIN10TH2)
+            ULONG ProcessReadyQueue : 1;
+#endif
+            ULONG ProcessReadyQueue : 1;
+            ULONG WaitNext : 1;
+            ULONG SystemAffinityActive : 1;
+            ULONG Alertable : 1;
+#if (NTDDI_VERSION < NTDDI_WIN81)
+            ULONG CodePatchInProgress : 1;
+#endif
+            ULONG UserStackWalkActive : 1;
+            ULONG ApcInterruptRequest : 1;
+            ULONG QuantumEndMigrate : 1;
+            ULONG UmsDirectedSwitchEnable : 1;
+            ULONG TimerActive : 1;
+            ULONG SystemThread : 1;
+            ULONG ProcessDetachActive : 1;
+            ULONG CalloutActive : 1;
+            ULONG ScbReadyQueue : 1;
+            ULONG ApcQueueable : 1;
+            ULONG ReservedStackInUse : 1;
+            ULONG UmsPerformingSyscall : 1;
+            ULONG DisableStackCheck : 1;
+            ULONG Reserved : 12;
+        };
+        LONG MiscFlags;
+    };
+
+    union
+    {
+        struct
+        {
+            ULONG AutoAlignment : 1;
+            ULONG DisableBoost : 1;
+            ULONG UserAffinitySet : 1;
+            ULONG AlertedByThreadId : 1;
+            ULONG QuantumDonation : 1;
+            ULONG EnableStackSwap : 1;
+            ULONG GuiThread : 1;
+            ULONG DisableQuantum : 1;
+            ULONG ChargeOnlyGroup : 1;
+            ULONG DeferPreemption : 1;
+            ULONG QueueDeferPreemption : 1;
+            ULONG ForceDeferSchedule : 1;
+            ULONG ExplicitIdealProcessor : 1;
+            ULONG FreezeCount : 1;
+#if (NTDDI_VERSION >= 0x060324D7) // since 6.3.9431.0
+            ULONG TerminationApcRequest : 1;
+#endif
+#if (NTDDI_VERSION >= 0x06032580) // since 6.3.9600.16384
+            ULONG AutoBoostEntriesExhausted : 1;
+#endif
+#if (NTDDI_VERSION >= 0x06032580) // since 6.3.9600.17031
+            ULONG KernelStackResident : 1;
+#endif
+#if (NTDDI_VERSION >= NTDDI_WIN10)
+            ULONG CommitFailTerminateRequest : 1;
+            ULONG ProcessStackCountDecremented : 1;
+            ULONG ThreadFlagsSpare : 5;
+#endif
+            ULONG EtwStackTraceApcInserted : 8;
+#if (NTDDI_VERSION < NTDDI_WIN10)
+            ULONG ReservedFlags : 10;
+#endif
+        };
+        LONG ThreadFlags;
+    };
+
+#if (NTDDI_VERSION >= NTDDI_WIN10)
+    volatile UCHAR Tag;
+    UCHAR SystemHeteroCpuPolicy;
+    UCHAR UserHeteroCpuPolicy : 7;
+    UCHAR ExplicitSystemHeteroCpuPolicy : 1;
+    UCHAR Spare0;
+#else
+    ULONG Spare0;
+#endif
+    ULONG SystemCallNumber;
+#ifdef _WIN64
+    ULONG Spare1; // Win 10: Spare10 ?
+#endif
+    PVOID FirstArgument;
+    PKTRAP_FRAME TrapFrame;
+
+    union
+    {
+        KAPC_STATE ApcState;
+        struct
+        {
+            UCHAR ApcStateFill[FIELD_OFFSET(KAPC_STATE, UserApcPending) + 1]; // 32bit: , 64bit: 43/0x2B
```

Comment:
Do you have RTL_SIZEOF_THROUGH_FIELD available here?
`RTL_SIZEOF_THROUGH_FIELD(KAPC_STATE, UserApcPending)` seems more logical